wgpu-types/src/lib.rs

```diff
@@ -7995,155 +7995,108 @@ pub enum DeviceLostReason {
     Destroyed = 1,
 }
 
-/// Descriptor for creating a shader module.
-///
-/// This type is unique to the Rust API of `wgpu`. In the WebGPU specification,
-/// only WGSL source code strings are accepted.
+/// Descriptor for a shader module given by any of several sources, with optional reflection information.
+/// At least one shader type that may be used by the backend must be `Some` or a panic is raised.
 #[derive(Debug, Clone)]
-pub enum CreateShaderModuleDescriptorPassthrough<'a, L> {
-    /// Passthrough for SPIR-V binaries.
-    SpirV(ShaderModuleDescriptorSpirV<'a, L>),
-    /// Passthrough for MSL source code.
-    Msl(ShaderModuleDescriptorMsl<'a, L>),
-    /// Passthrough for DXIL compiled with DXC
-    Dxil(ShaderModuleDescriptorDxil<'a, L>),
-    /// Passthrough for HLSL
-    Hlsl(ShaderModuleDescriptorHlsl<'a, L>),
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct CreateShaderModuleDescriptorPassthrough<'a, L> {
+    /// Entrypoint. Unused for Spir-V.
+    pub entry_point: String,
+    /// Debug label of the shader module. This will show up in graphics debuggers for easy identification.
+    pub label: L,
+    /// Number of workgroups in each dimension x, y and z. Unused for Spir-V.
+    pub num_workgroups: (u32, u32, u32),
+    /// Optional reflection information (currently empty and meaningless).
+    pub reflection: Option<ShaderModuleReflection>,
+    /// Runtime checks that should be enabled.
+    pub runtime_checks: ShaderRuntimeChecks,
+
+    /// Binary SPIR-V data, in 4-byte words.
+    pub spirv: Option<Cow<'a, [u32]>>,
+    /// Shader DXIL source.
+    pub dxil: Option<Cow<'a, [u8]>>,
+    /// Shader MSL source.
+    pub msl: Option<Cow<'a, str>>,
+    /// Shader HLSL source.
+    pub hlsl: Option<Cow<'a, str>>,
+    /// Shader GLSL source (currently unused).
+    pub glsl: Option<Cow<'a, str>>,
+    /// Shader WGSL source (currently unused).
```

Comment:
This isn't true anymore.
```suggestion
    /// Shader WGSL source.
```